File: frontend/src/api/userConfig.ts
import service from './request'

// 用户配置请求参数
export interface UserConfigRequest {
  llm_api_key?: string
  llm_base_url?: string
  llm_model?: string
  xiaohongshu_cookie?: string
  xiaohongshu_user_id?: string
  xiaohongshu_token?: string
  default_publish_time?: string
  auto_publish_enabled?: boolean
}

// 用户配置响应
export interface UserConfigResponse {
  id: number
  user_id: number
  llm_api_key: string
  llm_base_url: string
  llm_model: string
  xiaohongshu_cookie: string
  xiaohongshu_user_id: string
  xiaohongshu_token: string
  default_publish_time: string
  auto_publish_enabled: boolean
  created_at: string
  updated_at: string
}

// 获取用户配置
export const getUserConfig = () => {
  return service.get<UserConfigResponse>('/user/config')
}

// 更新用户配置
export const updateUserConfig = (data: UserConfigRequest) => {
  return service.put<UserConfigResponse>('/user/config', data)
}

File: backend/internal/model/user_config.go
// Package model 定义数据模型
package model

// UserConfigRequest 用户配置请求（用于 API 请求）
// 注意：UserConfig 模型已迁移到独立的表中（llm_providers, xiaohongshu_configs, publish_configs）
// 此处保留 Request 类型用于 API 请求兼容
type UserConfigRequest struct {
	// LLM 配置
	LLMAPIKey  string `json:"llm_api_key"`
	LLMBaseURL string `json:"llm_base_url"`
	LLMModel   string `json:"llm_model"`

	// 小红书配置
	XiaohongshuCookie string `json:"xiaohongshu_cookie"`
	XiaohongshuUserId string `json:"xiaohongshu_user_id"`
	XiaohongshuToken  string `json:"xiaohongshu_token"`

	// 发布配置
	DefaultPublishTime string `json:"default_publish_time"`
	AutoPublishEnabled bool   `json:"auto_publish_enabled"`
}
